synapse/storage/background_updates.py

```diff
@@ -122,6 +122,8 @@ def get_current_update(self) -> Optional[BackgroundUpdatePerformance]:
 
     def start_doing_background_updates(self) -> None:
         if self.enabled:
+            # if we start a new background update, not all updates ar done
```

Comment:
```suggestion
            # if we start a new background update, not all updates are done.
```

Does this fix a bug or just an oddity?

Reply:
IMO this is a bug.
I am wrote the unit tests first. I did not get the right status with `has_completed_background_updates`. After finish all jobs after synapse start, `_all_done` is always `True`. There is no code to set it `False` if there is a new job.

Reply:
My question was poorly worded -- I agree it is a bug! Do you think it has any impact in normal operations or properly not?

Reply:
IMO it has no other impact. `_all_done` is only analysed in `has_completed_background_updates` and `has_completed_background_update` to skip db query. It should always return the right status. Most calls are in unit tests.

Reply:
Thanks for the analysis, that was my gut feeling too!